task-sdk/src/airflow/sdk/definitions/deadline_reference.py

```diff
@@ -0,0 +1,136 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any
+
+from airflow.models.deadline import _fetch_from_db
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+
+class BaseDeadlineReference(LoggingMixin, ABC):
+    """Base class for all Deadline implementations."""
+
+    # Whether the evaluation requires conditions.
+    requires_conditions = True
+
+    # Set of required kwargs - subclasses should override this.
+    required_kwargs: set[str] = set()
+
+    def evaluate_with(self, **kwargs: Any) -> datetime:
+        """Validate the provided kwargs and evaluate this deadline with the given conditions."""
+        missing_kwargs = self.required_kwargs - set(kwargs.keys())
+        if missing_kwargs:
+            raise ValueError(f"Missing required parameters: {', '.join(missing_kwargs)}")
```

Comment:
```suggestion
        if missing_kwargs := {k for k in self.required_kwargs if k not in kwargs}:
            raise ValueError(f"Missing required parameters: {', '.join(missing_kwargs)}")
```

Reply:
I've added more robust kwarg verification both ways (missing and extra), thanks for the suggestion